engine/api/src/main/java/io/deephaven/engine/table/ColumnUpdateClause.java

```diff
@@ -0,0 +1,30 @@
+package io.deephaven.engine.table;
+
+import io.deephaven.engine.table.updateBySpec.UpdateBySpec;
+import org.jetbrains.annotations.NotNull;
+
+public class ColumnUpdateClause implements UpdateByClause {
+    private final UpdateBySpec spec;
+    @NotNull
+    private final String[] columns;
```

Comment:
Are these pure column names? Or, can we expect expressions here?

Might be tighter to use `io.deephaven.api.ColumnName`. Also, if making immutable, should be `List` instead of `[]`.

Reply:
I think we could use `ColumnName` but I can't manage getting the namespaces sorted to use them